### PR TITLE
Add and fix API links in plot examples

### LINKED ISSUE #5111: More complete linking of API methods in Gallery Examples
### Describe what maintenance you would like added.

The gallery examples are extremely useful to help users get started.  But, often I want to jump from the example to the detailed documentation for the particular method that is used in the example.  An example of a gallery example that does this well is https://docs.pyvista.org/version/stable/examples/01-filter/clipping.html, where links to `clip`, `clip_box` and so on are provided.  An example of a gallery example that does not currently do this well is https://docs.pyvista.org/version/stable/examples/01-filter/project-plane.html where there is no link to `project_points_to_plane`.

### Links to source code.

Example that links to detailed documentation:
https://github.com/pyvista/pyvista/blob/v0.42.3/examples/01-filter/clipping.py

Example that doesn't link to detailed documentation:
https://github.com/pyvista/pyvista/blob/v0.42.3/examples/01-filter/project-plane.py

### Pseudocode or Screenshots

_No response_

### PR DESCRIPTION
**Pull request by alioacar**
resolve #5111.

1. Added  API links to:
   - Several plot examples in `examples/02-plot/`

2. Fixed incorrect API links:
   - `examples/02-plot/legend.py`

3. Added a short explanation and API link to:
   - `examples/02-plot/linked.py`

### CODE REVIEW

File: examples/02-plot/anti-aliasing.py [hunk 1]
```diff
@@ -41,8 +41,8 @@
 # %%
 # No Anti-Aliasing
 # ~~~~~~~~~~~~~~~~
-# First, let's show a plot without any anti-aliasing.
-
+# First, let's show a plot without any anti-aliasing
+# using :func:`disable_anti_aliasing() <pyvista.Plotter.disable_anti_aliasing>`.
```
Comment (user27182) [inline examples/02-plot/anti-aliasing.py:45]:
```suggestion
# using :func:`~pyvista.Plotter.disable_anti_aliasing`.
```

File: examples/02-plot/anti-aliasing.py [hunk 2]
```diff
@@ -71,6 +71,7 @@
 
 # %%
 # You can increase the smoothing by increasing multi_samples
+# using :func:`enable_anti_aliasing() <pyvista.Plotter.enable_anti_aliasing>`.
```
Comment (user27182) [inline examples/02-plot/anti-aliasing.py:74]:
```suggestion
# using :func:`~pyvista.Plotter.enable_anti_aliasing`.
```

File: examples/02-plot/chart_basics.py [hunk 3]
```diff
@@ -23,9 +23,9 @@
 
 # %%
 # This example shows how to create a 2D scatter plot from 100 randomly sampled
-# datapoints. By default, the chart automatically rescales its axes such that
-# all plotted data is visible. By right clicking on the chart you can enable
-# zooming and panning of the chart.
+# datapoints using :func:`scatter() <pyvista.Chart2D.scatter>`. By default, the chart automatically
```
Comment (user27182) [inline examples/02-plot/chart_basics.py:26]:
```suggestion
# datapoints using :func:`~pyvista.Chart2D.scatter`. By default, the chart automatically
```

File: examples/02-plot/chart_basics.py [hunk 4]
```diff
@@ -35,8 +35,8 @@
 
 # %%
 # To connect datapoints with lines, you can create a 2D line plot as shown in
-# the example below. You can also dynamically 'zoom in' on the plotted data
-# by specifying a custom axis range yourself.
+# the example below using :func:`line() <pyvista.Chart2D.line>`. You can also dynamically
```
Comment (user27182) [inline examples/02-plot/chart_basics.py:38]:
```suggestion
# the example below using :func:`~pyvista.Chart2D.line`. You can also dynamically
```

File: examples/02-plot/chart_basics.py [hunk 5]
```diff
@@ -58,7 +58,8 @@
 chart.show()
 
 # %%
-# The following example shows how to create filled areas between two polylines.
+# The following example shows how to create filled areas between two polylines
+# using :func:`area() <pyvista.Chart2D.area>`.
```
Comment (user27182) [inline examples/02-plot/chart_basics.py:62]:
```suggestion
# using :func:`~pyvista.Chart2D.area`.
```

File: examples/02-plot/chart_basics.py [hunk 6]
```diff
@@ -71,8 +72,8 @@
 chart.show()
 
 # %%
-# Bar charts are also supported. Multiple bar plots are placed next to each
-# other.
+# Bar charts are also supported using :func:`bar() <pyvista.Chart2D.bar>`.
```
Comment (user27182) [inline examples/02-plot/chart_basics.py:75]:
```suggestion
# Bar charts are also supported using :func:`~pyvista.Chart2D.bar`.
```

File: examples/02-plot/chart_basics.py [hunk 7]
```diff
@@ -128,15 +130,16 @@
 # %%
 # Beside the flexible Chart2D used in the previous examples, there are a couple
 # other dedicated charts you can create. The example below shows how a pie
-# chart can be created.
+# chart can be created using :class:`ChartPie <pyvista.ChartPie>`.
```
Comment (user27182) [inline examples/02-plot/chart_basics.py:133]:
```suggestion
# chart can be created using :class:`~pyvista.ChartPie`.
```

File: examples/02-plot/chart_basics.py [hunk 8]
```diff
@@ -128,15 +130,16 @@
...
 
 data = np.array([8.4, 6.1, 2.7, 2.4, 0.9])
 chart = pv.ChartPie(data)
 chart.plot.labels = [f'slice {i}' for i in range(len(data))]
 chart.show()
 
 # %%
-# To summarize statistics of datasets, you can easily create a boxplot.
+# To summarize statistics of datasets, you can easily create a boxplot
+# using :class:`ChartBox <pyvista.ChartBox>`.
```
Comment (user27182) [inline examples/02-plot/chart_basics.py:142]:
```suggestion
# using :class:`~pyvista.ChartBox`.
```

File: examples/02-plot/depth-peeling.py [hunk 9]
```diff
@@ -9,7 +9,7 @@
 have issues with this routine.
 
 For this example, we will showcase the difference that depth peeling
-provides.
+provides. See :func:`enable_depth_peeling() <pyvista.Plotter.enable_depth_peeling>`.
```
Comment (user27182) [inline examples/02-plot/depth-peeling.py:12]:
```suggestion
provides. See :func:`~pyvista.Plotter.enable_depth_peeling`.
```

File: examples/02-plot/floors.py [hunk 10]
```diff
@@ -4,7 +4,8 @@
 Plot with Floors
 ~~~~~~~~~~~~~~~~
 
-Add a floor/wall at the boundary of the rendering scene.
+Add a floor/wall at the boundary of the rendering scene
+using :func:`add_floor() <pyvista.Plotter.add_floor>`.
```
Comment (user27182) [inline examples/02-plot/floors.py:8]:
```suggestion
using :func:`~pyvista.Plotter.add_floor`.
```

File: examples/02-plot/labels.py [hunk 11]
```diff
@@ -38,7 +38,7 @@
 poly
 
 # %%
-# Now plot the points with labels:
+# Now plot the points with labels using :func:`add_point_labels() <pyvista.Plotter.add_point_labels>`.
```
Comment (user27182) [inline examples/02-plot/labels.py:41]:
```suggestion
# Now plot the points with labels using :func:`~pyvista.Plotter.add_point_labels`.
```

File: examples/02-plot/legend.py [hunk 12]
```diff
@@ -21,7 +21,7 @@
 # Default legend for glyphs
 # +++++++++++++++++++++++++
 #
-# The method :func:`pyvista.PolyData.add_legend` is able to retrieve and use
+# The method :func:`add_legend() <pyvista.Plotter.add_legend>` is able to retrieve and use
```
Comment (user27182) [inline examples/02-plot/legend.py:24]:
```suggestion
# The method :func:`~pyvista.Plotter.add_legend` is able to retrieve and use
```

File: examples/02-plot/legend.py [hunk 13]
```diff
@@ -52,7 +52,7 @@
 # Using custom legends
 # ++++++++++++++++++++
 #
-# You can use specific labels with :func:`pyvista.PolyData.add_legend`
+# You can use specific labels with :func:`add_legend() <pyvista.Plotter.add_legend>`
```
Comment (user27182) [inline examples/02-plot/legend.py:55]:
```suggestion
# You can use specific labels with :func:`~pyvista.Plotter.add_legend`
```

File: examples/02-plot/legend.py [hunk 14]
```diff
@@ -71,7 +71,7 @@
 pl.add_legend(legend)
 
 # In this case, the default values are used, not the ones from the
-# :func:`pyvista.PolyData.add_mesh`.
+# :func:`add_mesh() <pyvista.Plotter.add_mesh>`.
```
Comment (user27182) [inline examples/02-plot/legend.py:74]:
```suggestion
# :func:`~pyvista.Plotter.add_mesh`.
```

File: examples/02-plot/linked.py [hunk 15]
```diff
@@ -4,6 +4,11 @@
 Linked Views in Subplots
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
+This example demonstrates how to create linked views in PyVista subplots
+using :func:`link_views() <pyvista.Plotter.link_views>`, where camera movements
```
Comment (user27182) [inline examples/02-plot/linked.py:8]:
```suggestion
using :func:`~pyvista.Plotter.link_views`, where camera movements
```

File: examples/02-plot/orbit.py [hunk 16]
```diff
@@ -30,7 +30,8 @@
 mesh = examples.download_st_helens().warp_by_scalar()
 
 # %%
-# Orbit around the Mt. St Helens dataset.
+# Orbit around the Mt. St Helens dataset using
+# :func:`generate_orbital_path() <pyvista.Plotter.generate_orbital_path>`.
```
Comment (user27182) [inline examples/02-plot/orbit.py:34]:
```suggestion
# :func:`~pyvista.Plotter.generate_orbital_path`.
```

File: examples/02-plot/point-clouds.py [hunk 17]
```diff
@@ -149,7 +149,7 @@
 # Orbit a Point Cloud
 # ~~~~~~~~~~~~~~~~~~~
 # Generate a plot orbiting around a point cloud. Color based on the distance
-# from the center of the cloud.
+# from the center of the cloud using :func:`generate_orbital_path() <pyvista.Plotter.generate_orbital_path>`.
```
Comment (user27182) [inline examples/02-plot/point-clouds.py:152]:
```suggestion
# from the center of the cloud using :func:`~pyvista.Plotter.generate_orbital_path`.
```

File: examples/02-plot/scalar-bars.py [hunk 18]
```diff
@@ -23,8 +23,8 @@
 # %%
 # By default, when plotting a dataset with a scalar array, a scalar bar for that
 # array is added. To turn off this behavior, a user could specify
-# ``show_scalar_bar=False`` when calling ``.add_mesh()``. Let's start with a
-# sample dataset provide via PyVista to demonstrate the default behavior of
+# ``show_scalar_bar=False`` when calling :func:`add_mesh() <pyvista.Plotter.add_mesh>`.
```
Comment (user27182) [inline examples/02-plot/scalar-bars.py:26]:
```suggestion
# ``show_scalar_bar=False`` when calling :func:`~pyvista.Plotter.add_mesh`.
```

File: examples/02-plot/ssao.py [hunk 19]
```diff
@@ -46,7 +46,7 @@
 # %%
 # Plot with SSAO
 # ~~~~~~~~~~~~~~
-# Now plot this with SSAO. Note how adjacent cubes affect the lighting of each
+# Now plot this with SSAO using :func:`enable_ssao() <pyvista.Plotter.enable_ssao>`. Note how adjacent cubes affect the lighting of each
```
Comment (user27182) [inline examples/02-plot/ssao.py:49]:
```suggestion
# Now plot this with SSAO using :func:`~pyvista.Plotter.enable_ssao`. Note how adjacent cubes affect the lighting of each
```

**Review by user27182 — COMMENTED** [review]
Thanks for the updates!
I suggest using `~` to shorten the links. See **Modified domain reference** in https://www.sphinx-doc.org/en/master/usage/referencing.html#syntax

File: examples/02-plot/chart_basics.py [hunk 20]
```diff
@@ -47,7 +47,7 @@
 
 # %%
 # You can also easily combine scatter and line plots using the general
-# :func:`pyvista.Chart2D.plot` function, specifying both the line and marker
+# :func:`plot() <pyvista.Chart2D.plot>` function, specifying both the line and marker
```
Comment (user27182) [inline examples/02-plot/chart_basics.py:50]:
```suggestion
# :func:`~pyvista.Chart2D.plot` function, specifying both the line and marker
```

File: examples/02-plot/chart_basics.py [hunk 21]
```diff
@@ -116,7 +117,8 @@
 chart.show()
 
 # %%
-# In a similar way, you can stack multiple area plots on top of each other.
+# In a similar way, you can stack multiple area plots on top of
+# each other using :func:`stack() <pyvista.Chart2D.stack>`.
```
Comment (user27182) [inline examples/02-plot/chart_basics.py:121]:
```suggestion
# each other using :func:`~pyvista.Chart2D.stack`.
```

**Review by user27182 — APPROVED** [review]
A few more formatting suggestions but otherwise LGTM, thanks!

**Comment by alioacar** [comment]
Thanks for the corrections!

**Review by tkoyama010 — APPROVED** [review]

**Pull request merged**